src/app.rb

```diff
@@ -0,0 +1,144 @@
+    else
+      puts "\nMust enter title and author to create a book"
+    end
+  end
+
+  def init_books_and_persons(db_books, db_persons)
+    puts "\nBOOKS AND PERSONS ADDED !"
+    db_books << Book.new('The King', 'Richard Harris')
+    db_books << Book.new('Sophie', 'Mark Twain')
+    db_books << Book.new('The Cure', 'Fred Winnipeg')
+
+    db_persons << Teacher.new('Biology', 41, 'Martha Ross')
+    db_persons << Teacher.new('History', 33, 'John Matheus')
+
+    db_persons << Student.new(21, 'Richard', parent_permission: true)
+    db_persons << Student.new(14, 'Mary', parent_permission: false)
+  end
+
+  def create_person(db_persons)
+    print "\nDo you want to create a student (1) or a teacher (2)? [Input the number]: "
+    person_type = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+
+    case person_type
+    when '2'
+      create_teacher(db_persons, name, age)
+    when '1'
+      create_student(db_persons, name, age)
+    else
+      puts "\nInvalid person type selected"
+    end
+  end
+
+  def create_teacher(db_persons, name, age)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    db_persons << teacher
+    puts "\nTeacher created successfully"
+  end
+
+  def create_student(db_persons, name, age)
+    print 'Has parent permission? [Y/N]: '
+    has_permission = gets.chomp.downcase
+    student = ''
+    case has_permission
+    when 'y'
+      student = Student.new(age, name, parent_permission: true)
+    when 'n'
+      student = Student.new(age, name, parent_permission: false)
+    else
+      puts "\ninvalid selection for permission"
+      return
+    end
+    db_persons << student
+    puts "\nStudent created successfully"
+  end
+
+  def create_rental(db_data)
+    if db_data[:books].empty? == true || db_data[:persons].empty? == true
+      puts "\nNo books or people registered for rent"
+      return
+    end
+
+    puts "\nSelect a book from the following list by number"
+
+    db_data[:books].each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+
+    book_index = gets.chomp.to_i
+    book_selected = db_data[:books][book_index]
+
+    puts "\nSelect a person from the following list by number"
+
+    db_data[:persons].each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    person_selected = db_data[:persons][person_index]
+
+    print 'Date (yyyy/mm/dd): '
+    date = gets.chomp
+
+    person_selected.add_rental(book_selected, date)
+    puts "\nRental created successfully for #{person_selected.name}"
+  end
+
+  def list_rentals(db_persons)
+    if db_persons.empty?
+      puts "\nNo one has rented yet"
+      return
+    end
+    print "\nID of the person: "
+    person_id = gets.chomp.to_i
```

Comment:
**[Optional]**

- [ ] It would be nice if you show person names with their ids because we cannot memorize id of a person but his/her name so it would be easy for us to see the id of that person if their will be the name of that person as well :blush: before inserting the id to show the rentals :blush:

Reply:
@HAMMAS-SALEEM good idea ! 
I 'll do it in the next days 👍 

Thank you for your review !